boards/banglejs-2/board.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2022 Nicole Faerber, Siegen, Germany
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_baglejs-2
+ * @{
+ *
+ * @file
+ * @brief       Board initialization for the Bagle JS2
+ *
+ * @author      Nicole Faebrer <nicole.faerber@digitaluhr-manufaktur.de>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "board.h"
+#include "mtd.h"
+#include "mtd_spi_nor.h"
+
+#include "periph/gpio.h"
+#include "periph/spi.h"
+#include "periph/pm.h"
+#include "periph/adc.h"
+#include "periph/pwm.h"
+#include "periph/spi.h"
+
+#include "timex.h"
+#include "ztimer.h"
+
+
+#ifdef MODULE_MTD
+static const mtd_spi_nor_params_t _banglejs2_nor_params = {
+    .opcode = &mtd_spi_nor_opcode_default,
+    .wait_chip_erase = 9LU * US_PER_SEC,
+    .wait_32k_erase = 160LU *US_PER_MS,
+    .wait_sector_erase = 70LU * US_PER_MS,
+    .wait_chip_wake_up = 1LU * US_PER_MS,
+    .clk = BANGLEJS2_NOR_SPI_CLK,
+    .flag = BANGLEJS2_NOR_FLAGS,
+    .spi = BANGLEJS2_NOR_SPI_DEV,
+    .mode = BANGLEJS2_NOR_SPI_MODE,
+    .cs = BANGLEJS2_NOR_SPI_CS,
+    .wp = GPIO_UNDEF,
+    .hold = GPIO_UNDEF,
+};
+
+static mtd_spi_nor_t banglejs2_nor_dev = {
+    .base = {
+        .driver = &mtd_spi_nor_driver,
+        .page_size = BANGLEJS2_NOR_PAGE_SIZE,
+        .pages_per_sector = BANGLEJS2_NOR_PAGES_PER_SECTOR,
+    },
+    .params = &_banglejs2_nor_params,
+};
+
+mtd_dev_t *mtd0 = (mtd_dev_t *)&banglejs2_nor_dev;
+#endif /* MODULE_MTD */
```

Comment:
If you want to have a file system on the external flash you could add

```C
#ifdef MODULE_VFS_DEFAULT
#include "vfs_default.h"
VFS_AUTO_MOUNT(littlefs2, VFS_MTD(banglejs2_nor_dev), VFS_DEFAULT_NVM(0), 0);
#endif
```

here and 

```make
ifneq (,$(filter vfs_default,$(USEMODULE)))
  USEPKG += littlefs2
  USEMODULE += mtd
endif
```

to `Makefile.dep`